### PR TITLE
Camel Main standalone can now more easily be configured

### PR DESCRIPTION
**Pull request by davsclaus**
This is also needed by Camel K to make it easier to configure and use Camel K with Java.

### CODE REVIEW

**Comment by davsclaus** [comment]
The example is likely better to view in isolation at
https://github.com/apache/camel/tree/bind/examples/camel-example-main

File: platforms/spring-boot/spring-boot-dm/camel-spring-boot-dependencies/pom.xml [hunk 1]
```diff
@@ -3431,7 +3431,7 @@
       <dependency>
         <groupId>org.apache.mina</groupId>
         <artifactId>mina-core</artifactId>
-        <version>2.0.17</version>
+        <version>2.0.20</version>
```
Comment (oscerd) [inline platforms/spring-boot/spring-boot-dm/camel-spring-boot-dependencies/pom.xml]: Did you check the Karaf feature :-)

File: pom.xml [hunk 2]
```diff
@@ -61,7 +61,7 @@
     <errorprone-version>2.5</errorprone-version>
     <mycila-license-version>3.0</mycila-license-version>
     <gmavenplus-plugin-version>1.6.2</gmavenplus-plugin-version>
-    <groovy-version>2.5.5</groovy-version>
+    <groovy-version>2.5.6</groovy-version>
```
Comment (oscerd) [inline pom.xml]: Did you check the Karaf feature? :-)

File: components/camel-properties/src/main/java/org/apache/camel/component/properties/DefaultPropertiesParser.java [hunk 3]
```diff
@@ -318,7 +318,14 @@ private String doGetPropertyValue(String key) {
                 }
             }
             if (value == null && envMode == PropertiesComponent.ENVIRONMENT_VARIABLES_MODE_OVERRIDE) {
-                value = System.getenv(key);
+                // lookup OS env with upper case key
```
Comment (dmvolod) [inline components/camel-properties/src/main/java/org/apache/camel/component/properties/DefaultPropertiesParser.java]: @davsclaus , what do you think about to move common code to the separate function?

Comment (davsclaus) [reply components/camel-properties/src/main/java/org/apache/camel/component/properties/DefaultPropertiesParser.java]: Yeah we can do that

**Comment by davsclaus** [comment]
Thanks for the review

**Pull request merged**